pyteal/ast/abi/address.py

```diff
@@ -15,19 +24,56 @@ def new_instance(self) -> "Address":
     def __str__(self) -> str:
         return "address"
 
+    def __eq__(self, other: object) -> bool:
+        return isinstance(other, AddressTypeSpec)
+
 
 AddressTypeSpec.__module__ = "pyteal"
 
 
 class Address(StaticArray):
     def __init__(self) -> None:
-        super().__init__(AddressTypeSpec(), ADDRESS_LENGTH)
+        super().__init__(AddressTypeSpec())
```

Comment:
If I am understanding correctly, you are trying to init `Address` object on a chain of inheritance:
```
BaseType -> StaticArray -> Address
```
A direct `super` method only allow you to inherit from `StaticArray`, which requires you to provide:
- argument TypeSpec
- array length

One way to get around is: init from grandparent object, e.g., from this link: https://stackoverflow.com/questions/62744176/how-to-overwrite-parent-class-init-method-and-use-super-to-call-grandparent-ini

Such technique would allow you to cause smaller change to interfaces for classes including:
- DynamicArray
- StaticArray
- Tuple

Reply:
Good point, that would work, but it's a little hacky.

I think this change is acceptable since we'll likely want to support more subtypes in the future, so might as well make it easier now.

The only downside is that it makes it more cumbersome for users to create objects. But in my opinion it's already far from ideal, so a little worse doesn't hurt that much.

To really streamline things, I think we could provide a Go-inspired `make` function like this:

```python
from typing import TypeVar
from pyteal import *

T = TypeVar("T", bound=abi.BaseType)

def make(t: type[T]) -> T:
    return abi.type_spec_from_annotation(t).new_instance()

myTuple = make(abi.Tuple2[abi.Uint32, abi.Uint64])
```

Reply:
Is this something I should implement in this PR?